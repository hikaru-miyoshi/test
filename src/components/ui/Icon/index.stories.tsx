import { ComponentMeta, ComponentStory } from "@storybook/react";
import React from "react";

import Icon from "./index";

export default {
  title: "Components/Icon",
  component: Icon,
  argTypes: {
    src: {
      control: "text",
      defaultValue: "/icon/heart.png",
    },
    alt: {
      control: "text",
      defaultValue: "heart",
    },
  },
} as ComponentMeta<typeof Icon>;

const Template: ComponentStory<typeof Icon> = args => <Icon {...args} />;

export const Basic = Template.bind({});
