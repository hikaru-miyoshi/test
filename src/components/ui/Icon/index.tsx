import classnames from "classnames/bind";
import { ComponentPropsWithoutRef, FC, memo } from "react";

import styled from "./style.module.scss";

const cx = classnames.bind(styled);

type Props = ComponentPropsWithoutRef<"img">;

const Icon = memo(props => {
  const { className, src, alt, ...attr } = props;
  const classes = cx(styled.img, className);

  return (
    <>
      {src && (
        <div className={classes}>
          <img src={src} alt={alt} {...attr} />
        </div>
      )}
    </>
  );
}) as FC<Props>;

export default Icon;
