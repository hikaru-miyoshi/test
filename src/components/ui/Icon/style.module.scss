@use "@/styles/helper" as *;

$color-grey01: #bbb;

.img {
  align-items: center;
  justify-content: center;
  width: auto;
  height: 1em;
}
