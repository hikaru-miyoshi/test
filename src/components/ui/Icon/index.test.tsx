import { render } from "@testing-library/react";
import { expect, test } from "vitest";

import Icon from "./index";

test("Icon", () => {
  const { container } = render(<Icon src={"/icon/heart.png"} alt="heart" />);

  expect(container).toMatchSnapshot();
});
