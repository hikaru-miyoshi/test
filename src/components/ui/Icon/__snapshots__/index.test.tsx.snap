// Vitest Snapshot v1

exports[`Icon 1`] = `
<div>
  <div
    class="__img_97e405_97e405"
  >
    <img
      alt="heart"
      src="/icon/heart.png"
    />
  </div>
</div>
`;
